src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -69,21 +70,47 @@ public bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs
             return false;
         }
 
-        public async Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, InvocationReasons reasons, CancellationToken cancellationToken)
+        public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, InvocationReasons reasons, CancellationToken cancellationToken)
         {
-            Contract.ThrowIfFalse(document.IsFromPrimaryBranch());
-
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (!document.Project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
+            if (!project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
             {
                 return;
             }
 
+            var projectVersion = await project.GetDependentVersionAsync(cancellationToken).ConfigureAwait(false);
+
+            try
+            {
+                // Process all the documents in this project in parallel.  The project system can
+                // handle multiple calls into SetProjectItemDesignerTypeAsync and will batch them
+                // appropriately.
+                var tasks = new List<Task>();
+                foreach (var document in project.Documents)
+                {
+                    tasks.Add(AnalyzeDocumentAsync(projectVersion, document, cancellationToken));
```

Comment:
alternatively, we could pass 'tasks' into AnalyzerDocumentAsync, and have the analze method responsible for adding the 'SetProjectItemDesignerTypeAsync' task to it.   this would make us only parallel on that part of the process.

Reply:
Given that we can scan in OOP, i think parallel for the entire project seems like something reasonable for us.

Reply:
@heejaechang let me know if you have preferences here.